sdk/eventhub/azure-eventhub/azure/eventhub/_common.py

```diff
@@ -392,6 +417,20 @@ def size_in_bytes(self):
         """
         return self._size
 
+    @property
+    def starting_published_sequence_number(self):
```

Comment:
I still feel this needs a rename.... 
And I think my preference is to go with `published_sequence_number` have the documentation indicate that this refers to the first in the batch.

Reply:
IIRC batches are transactional right? There's no partial failure here?

Reply:
yes, batches are transactional, either all events in the batch get sent or failed.

Reply:
I don't have a strong preference on the name -- personally I think the "starting_" prefix adds some clarity for a batch object while with just `published_sequence_number` would give us the benefit of consistency with `EventData`.

@jsquire , would you be able to provide some context on the "starting_" prefix being chosen instead of just `starting_published_sequence_number`?

cc: @chradek @conniey

Reply:
"Starting" was chosen to denote that it was the sequence number applied to the first event in the batch.   The batch itself is not assigned a sequence number, nor is it considered a single item by the service.   

To determine the next sequence number that the service expects,  one has to take the batch's starting sequence number and increment it by the number of events that were in the batch.   This caused confusion for our beta users that expected the "LastPublishedSequenceNumber" returned as part of the partition publishing properties did not align with the "PublishedSequenceNumber" that we assigned to the batch.   It was agreed that the prefix "Starting" helped as a mnemonic.

Reply:
I have created a separate issue for further discussion on the namings: https://github.com/Azure/azure-sdk-for-python/issues/16994

Reply:
Yeah we can keep as-is for preview and settle before GA

Reply:
the preview may stay for a while (or even be pulled out) so we should have time to discuss among languages :P